17057 styled inplace edit

ulferts opened the pull request:
This styles the in-place edit functionality of the details pane as defined by the style guide. It does not:
- Style the inputs (text, select2, textarea). However it does style the toolbar above a textarea.
- Styles elements which are not yet editable. Therefore, there is a zigzag pattern in the attribute list for as long as not all elements are editable. That problem will solve itself once the element is editable.
- Styles the buttons for send, send&mail and cancel according to the style guide as it was changed after the buttons have been styled before. I will create a different package for that.
- Add functionality (I hope)

https://community.openproject.org/work_packages/17057
https://community.openproject.org/work_packages/18743
https://community.openproject.org/work_packages/18873
https://community.openproject.org/work_packages/19074
and in parts https://community.openproject.org/work_packages/19103

ulferts committed:
editing inplace edit for textareas to styleguide

ulferts committed:
adding in-place editor for single line to styleguide

ulferts committed:
provide more context to inplace edit in styleguide

ulferts committed:
adding inplace edit of multi line text to styleguide

ulferts committed:
inplace edit on text added to styleguide

ulferts committed:
removing angular leftovers

ulferts committed:
initial inplace edit trigger styling - breaks stuff

ulferts committed:
styles list of attributes that are editable

ulferts committed:
styling inplace edit actions area

ulferts committed:
styling inplace editing

ulferts committed:
style jstoolbar

ulferts committed:
renaming classes for consistency

ulferts committed:
remove unnecessary div

ulferts committed:
styling preview

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://ulferts-feature-styled-inplace-edit.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://ulferts-feature-styled-inplace-edit.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

myabc commented:
@ulferts I scanned through this very quickly and the approach / class structure looks good! :sunglasses:

NobodysNightmare commented:
The element around editable fields has a padding of 6px (`inplace-edit--read-value`?), where as readonly fields have no padding. This looks weird, especially, when none of the fields is highlighted:

![missing-padding](https://cloud.githubusercontent.com/assets/453584/6748507/ad0aa296-cee6-11e4-9bdf-6fd742d8f24b.PNG)

I think read-only fields (in the screenshot _percentage done_) should receive the same padding as editable fields.

edit: Just noticed your words about what you are not going to style here...

However note, that the problem will probably **_not solve itself**_, because there will remain situations where fields are not editable (hello `startDate` and `estimatedTime` on parent work packages!). Add to that fields that are not editable in any situation (hello `spentTime`!).

NobodysNightmare commented:
I like the look of the confirmation panel (e.g. on the subject). However, [OP #18743](https://community.openproject.org/work_packages/18743) clearly expects left-alignment and yours is right aligned.

![right-align](https://cloud.githubusercontent.com/assets/453584/6748952/bea07af8-ceeb-11e4-9f72-5724670254de.PNG)

Just asking...

P.S.: I'd go for colored error messages (how about red?), but I think someone made a concious decision that white is the red...

NobodysNightmare commented:
[OP #19074](https://community.openproject.org/work_packages/19074) does not seem fixed at all:

![long field name](https://cloud.githubusercontent.com/assets/453584/6748620/e4defcf2-cee7-11e4-92c6-e3f7b1692a18.PNG)

NobodysNightmare commented:
I just noticed something, that should not be too critical, but I want to mention it nevertheless.

The vertical alignment between the attribute title and the attribute value is slightly off, for both: editable fields and read-only fields. I say it is not critical, because I only noticed it on high zoom fatcors in my graphical editor.

![vertical-alignment](https://cloud.githubusercontent.com/assets/453584/6748753/a28eb9a8-cee9-11e4-8997-0cba380aaebd.PNG)

Is this something:
- [x] you want to immediately fix
- [ ] I should write down in a ticket
- [ ] we should forget and never talk about again

ulferts committed:
prevent scroll on attribute key label

ulferts committed:
fix vertical height disparity of key/value for wp attributes

ulferts committed:
define fixed height for edit trigger icon

ulferts commented:
@NobodysNightmare I addressed the issues you mentioned except for:
- I didn't style the error messages, that is outside the scope of this PR
- I didn't left align the like https://community.openproject.org/work_packages/18743 says because the specs differ.

NobodysNightmare commented:
Looks good from the users perspective :+1: 

I am not sure who will be reviewing the code side, I am probably not the best choice for that, but I could do it if neccessary...
